src/planetary_system.jl

```diff
@@ -352,6 +352,340 @@ function generate_period_and_sizes_power_law(s::Star, sim_param::SimParam; num_p
+        end
+        push!(P_segments_unstable, (P_lower, P_upper))
+    end
+    return P_segments_unstable
+end
+
+function compute_unstable_mutualHill_regions_periods_given_planets(P::AbstractVector{Float64}, mass::AbstractVector{Float64}, insert_pl_mass::Float64, star_mass::Float64, sim_param::SimParam; ecc::AbstractVector{Float64}=zeros(length(P)), insert_pl_ecc::Float64=0., verbose::Bool=false)
+    @assert length(P) == length(mass) == length(ecc)
+    min_num_mutual_hill_radii = get_real(sim_param, "num_mutual_hill_radii")
+    order = sortperm(P)
+
+    P_segments_unstable = Tuple{Float64,Float64}[]
+    for pl in 1:length(P)
+        a = semimajor_axis(P[order[pl]], star_mass)
+        mu = (mass[order[pl]] + insert_pl_mass)/star_mass
+        term = (min_num_mutual_hill_radii/2.)*(mu/3.)^(1//3)
+        @assert 0 < 1. - insert_pl_ecc - term # must be true or else inequality flips sign
+        a_lower, a_upper = a*(1. - ecc[order[pl]] - term)/(1. + insert_pl_ecc + term), a*(1. + ecc[order[pl]] + term)/(1. - insert_pl_ecc - term)
+        a_lower = max(a_lower, 0.) # set a_lower=0 if negative
+        @assert 0 <= a_lower < a_upper
+        P_lower, P_upper = period_given_semimajor_axis(a_lower, insert_pl_mass+star_mass), period_given_semimajor_axis(a_upper, insert_pl_mass+star_mass)
+        if verbose
+            println("P_blocked: ", (P_lower, P_upper))
+        end
+        push!(P_segments_unstable, (P_lower, P_upper))
+    end
+    return P_segments_unstable
+end
+
+function compute_unstable_mutualHill_regions_periodscales_given_clusters(pl_per_cl::Vector{Int64}, P_cl::AbstractVector{Float64}, mass_cl::AbstractVector{Float64}, insert_cl_ρ::AbstractVector{Float64}, insert_cl_mass::AbstractVector{Float64}, star_mass::Float64, sim_param::SimParam; ecc_cl::AbstractVector{Float64}=zeros(length(P_cl)), insert_cl_ecc::AbstractVector{Float64}=zeros(length(insert_cl_mass)), verbose::Bool=false)
+    n_cl = length(pl_per_cl)
+    @assert sum(pl_per_cl) == length(P_cl) == length(mass_cl) == length(ecc_cl)
+    @assert length(insert_cl_ρ) == length(insert_cl_mass) == length(insert_cl_ecc)
+    min_num_mutual_hill_radii = get_real(sim_param, "num_mutual_hill_radii")
+
+    # Get the inner-most and outer-most planets in the cluster we are trying to insert:
+    order_insert_cl = sortperm(insert_cl_ρ) # ρ denotes unscaled periods in this function
+    insert_pl_in_ρ, insert_pl_in_mass, insert_pl_in_ecc = insert_cl_ρ[order_insert_cl[1]], insert_cl_mass[order_insert_cl[1]], insert_cl_ecc[order_insert_cl[1]]
+    insert_pl_out_ρ, insert_pl_out_mass, insert_pl_out_ecc = insert_cl_ρ[order_insert_cl[end]], insert_cl_mass[order_insert_cl[end]], insert_cl_ecc[order_insert_cl[end]]
+    max_ratio = insert_pl_out_ρ/insert_pl_in_ρ
+
+    Pc_segments_unstable = Tuple{Float64,Float64}[]
+    pl_start = 1
+    pl_stop = 0
+    for c in 1:n_cl
+        pl_stop += pl_per_cl[c]
+        order = (pl_start - 1) .+ sortperm(P_cl[pl_start:pl_stop])
```

Comment:
Is this doing what you intend?  
Do you mean to access {P,mass,ecc}_cl[order[1]] and not {P,mass,ecc}_cl[pl_star:pl_stop][order[1]] ?

Reply:
This is the one I'd like you to look at most closely

Reply:
I think this is doing what I intend. I had updated this function multiple times to fix a series of bugs.
Initially, I was passing the planets in previous clusters (i.e. {P,mass,ecc}_cl) as arrays of vectors (i.e. a vector for each cluster, so Array{AbstractVector{Float64},1} objects), which worked well but got me worried about memory allocation (since in our models, we have vectors for the periods, masses, etc. of all the planets in a system, from all clusters, so I would have had to reshape things).
So now, my solution is to pass the planets from all the previous clusters as a single vector (one for P_cl, mass_cl, and ecc_cl), and additionally a vector "pl_per_cl" to keep track of how many planets are in each cluster. Thus, I've defined the "order" as "order = (pl_start - 1) .+ sortperm(P_cl[pl_start:pl_stop])" to index the planets in cluster c in order of increasing period.

Reply:
Ok.  Thanks, the docstring now explains what I was missing.